pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go

```diff
@@ -43,31 +43,22 @@ type AppManifests struct {
 	AppRevision     *v1beta1.ApplicationRevision
 	WorkloadOptions []WorkloadOption
 
-	appComponents []applicationComponent
-	components    []*v1alpha2.Component
-
-	appName        string
-	appNamespace   string
-	appLabels      map[string]string
-	appAnnotations map[string]string
-	appOwnerRef    *metav1.OwnerReference
+	componentManifests []*types.ComponentManifest
+	appName            string
+	appNamespace       string
+	appLabels          map[string]string
+	appAnnotations     map[string]string
+	appOwnerRef        *metav1.OwnerReference
 
 	assembledWorkloads map[string]*unstructured.Unstructured
 	assembledTraits    map[string][]*unstructured.Unstructured
 	// key is workload reference, values are the references of scopes the workload belongs to
-	referencedScopes map[runtimev1alpha1.TypedReference][]runtimev1alpha1.TypedReference
+	referencedScopes map[corev1.ObjectReference][]corev1.ObjectReference
```

Comment:
what's the difference between types in `runtimev1alpha1` and `corev1` here? I notice you still need to use `runtimev1alpha1` in this package.

Reply:
No difference indeed. I just think we are encouraged to use K8s libs instead of other 3rd-party libs. So I hope to avoid using libs from crossplane in the future.

Reply:
yes, good job